src/client/components/DatePicker/DatePicker.react.js

```diff
@@ -68,24 +58,10 @@ export default class DatePicker extends React.Component {
 
   render() {
     let {
-      showIcon,
       locale,
       ...inputProps
     } = this.props;
 
-    let element = (<input {...inputProps} ref="input"/>);
-
-    if (this.props.showIcon && !this.props.disabled) {
-      return (
-        <div className="input-group date" ref="group" >
-          {element}
-          <span className="input-group-addon">
-            <span className="glyphicon glyphicon-calendar"></span>
-          </span>
-        </div>
-      );
-    } else {
-      return element;
-    }
+    return (<input {...inputProps} ref="input"/>);
```

Comment:
the difference is that if we move ref before props then we will be able to override ref via props